### PR TITLE
Add new functionalities for sensor grid folder and receiver file

### PR DESCRIPTION
**Pull request by mostaphaRoudsari**
This PR:

1. Adds a command to provide support for adding output spec to a receiver file. This command is mainly useful to add the output spec to receiver files for view matrix calculation for 3/5 phase simulations.
2. Adds a new function for evenly distributing the sensors in a sensor grid folder. This command creates a new folder with the newly generated sensor grids.
3. Adds a new function for restructuring back the data in a folder based on dist_info. This function is useful to restructure the files that are generated based on the distributed sensors. This command is mainly useful after running the studies for the evenly distributed grids that are generated from the function in step 2.

I will add the two commands for these functions to honeybee-radiance CLI under the `grid` sub-command.

I can use some help with the names that I chose for the `distribution` and `restructuring` functions. I'm going to add them as `split-folder` and `merge-folder` to honeybee-radiance CLI to be consistent with `split` and `merge` options but they are not necessarily split and merge which was the case for the files.

cc: @ladybug-tools/radiance-developers

### CODE REVIEW

File: honeybee_radiance_folder/gridutil.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+import json
+
+
+def distribute_sensors(
+        input_folder, output_folder, grid_count, min_sensor_count=2000, verbose=False
+    ):
+    """Create new sensor grids folder with evenly distribute sensors.
+
+    This function creates a new folder with evenly distributed sensor grids. The folder
+    will include a ``_dist_info.json`` file which has the information to recreate the
```
Comment (chriswmackey) [inline honeybee_radiance_folder/gridutil.py:12]: I just had a comment on this choice of name, which isn't necessarily an issue but is maybe an indication that the name could be better. `dist-info` is the same name that PyPI uses to indicate the folders that contain the distribution information for each package:

image

... and when I first saw the words `dist-info`, I thought that this is what you were referring to. Granted, I don't think there's anything inaccurate with using the word "distribution" to describe what you are changing here about the sensor grids but maybe "redistribution" would be a bit clearer and would also help avoid the naming conflict with PyPI distributions. After all, the sensors are already distributed across sensor grids when you input them to this function and you are just "redistributing" them into equal-sized bins. That would make the name of the file `_redist_info.json`, which is different enough that I wouldn't have gotten confused in this case.

I also trust you if you want to use another name or you think these names are in such different spaces that only a handful of people like myself will get confused by them.

**Review by chriswmackey — COMMENTED** [review]
Thank you for this, @mostaphaRoudsari . I am very much looking forward to getting this merged.

All of my comments are about naming, spelling, and docstrings. So nothing major but I think it's worth taking some time to just consider a few naming options since they can become much more difficult to change once we make use of these methods in all grid-based recipes.

File: honeybee_radiance_folder/gridutil.py (hunk 1 above)
Comment (mostaphaRoudsari) [reply honeybee_radiance_folder/gridutil.py:12]: `_redist_info.json` it is!

File: honeybee_radiance_folder/gridutil.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+    print(
+        'Distributed %d sensors among %d grids with %d senseors each.' % (
+            total_count, grid_count, sensor_per_grid
+            )
+    )
+    return grid_count, sensor_per_grid
+
+
+def restructure_distributed_data(
```
[the remark this answers is not in the document]
Comment (mostaphaRoudsari) [reply honeybee_radiance_folder/gridutil.py:182]: I like `restore_original_distribution` more. Took that one.

**Pull request merged**